middleware/auth.js

```diff
@@ -102,18 +114,50 @@ const whoamiAuth = (req, res, next) => {
   return next("router")
 }
 
+// Replace access token with site access token if it is available
+const useSiteAccessTokenIfAvailable = async (req, _res, next) => {
+  const { authService, sitesService } = identityServices
+  const { accessToken: userAccessToken, userId } = req
+  const { siteName } = req.params
+
+  // Check if site is onboarded to Isomer identity, otherwise continue using user access token
+  const site = await sitesService.getBySiteName(siteName)
+  if (!site) {
+    logger.info(
+      `Site ${siteName} does not exists in site table. Default to using user access token.`
+    )
+    return next()
+  }
+
+  logger.info(`Verifying user's access to ${siteName}`)
+  if (!authService.hasAccessToSite(siteName, userId, userAccessToken)) {
+    throw new NotFoundError("Site does not exist")
+  }
+
+  const siteAccessToken = await sitesService.getSiteAccessToken(siteName)
+  req.accessToken = siteAccessToken
+  logger.info(
+    `User ${userId} has access to ${siteName}. Using site access token ${site.apiTokenName}.`
+  )
+
+  return next()
+}
+
 // Health check
 auth.get("/v2/ping", noVerify)
 
 // Login and logout
-auth.get("/v1/auth/github-redirect", noVerify)
 auth.get("/v1/auth", noVerify)
+auth.get("/v1/auth/github-redirect", noVerify)
 auth.delete("/v1/auth/logout", noVerify)
 auth.get("/v1/auth/whoami", whoamiAuth)
 
 // Index
 auth.get("/v1", noVerify)
 
+// Inject site access token if available
+auth.use("/v1/sites/:siteName", verifyJwt, useSiteAccessTokenIfAvailable)
+
```

Comment:
double check if it always hit any given endpoint which starts with this

Also add into v2?

Reply:
Update: it does